### ISSUE TITLE
API for creation of requests

### ISSUE THREAD

**Issue by mrsimpson** [RocketChat/Rocket.Chat#6889]
# Use case
Problems which are source of a discussion often occur in contexts when users do not actively work in Rocket.Chat, but in an external system (e. g. within a business system). Out of this context, a user shall be able to create a chat with colleagues who might know.

*Prerequisite: Support for requests and topics. (#6888)*

# Features

## API for creating a request
There shall be a http-endpoint to which an external system can post
* an initial question
* by a requestor
* an array of experts
* for an (optional) topic
* with (an optional) context of the source system

The URL of the create request-room shall be returned along with the members of the room.

## Creation of a request
Users shall be identified by eMail-Address.
* If no valid user for requestor could be found, the creation shall fail
* If not at least one valid user for an expert could be found, the creation shall fail
* A new request-room shall be created with owner requestor and all valid users passed as members
* the initial question shall be the first message of the channel
* The context shall be persisted along with the room - visualization is another story (literally)
* A human readable unique ID shall be generated (if a topic is passed using the usual prefixing+ID as usual for requests)

## Implicit update of topic
If a topic is passed to the API, a corresponding topic shall be created or updated with the members passed

# Why not an inbound integration
* Quite tricky to debug
* Tricky to deploy
* Re-Use
* Backend-Knowledge needed
  * counter for generate IDs
  * authorizations
  * Context-Model

**Comment by geekgonecrazy** [comment]
I'm not 100% sure what to do with this one.  What is a request?  Would this be like for the creation of a thread or a room of a certain type?

**Comment by mrsimpson** [comment]
Yup, it’s a creation of a particular roomtype with additional properties. 
I’d consider this not being part of the main application anymore though - it should be provided by a Rocketlet

**Comment by geekgonecrazy** [comment]
That was what I suspected :grin: thanks!
